pkg/provisioners/inmemory/channel/reconcile.go

```diff
@@ -130,133 +119,30 @@ func (r *reconciler) reconcile(ctx context.Context, c *eventingv1alpha1.Channel)
 
 	// We are syncing three things:
 	// 1. The K8s Service to talk to this Channel.
-	// 2. The Istio VirtualService to talk to this Channel.
 	// 3. The configuration of all Channel subscriptions.
 
-	// We always need to sync the Channel config, so do it first.
-	if err := r.syncChannelConfig(ctx); err != nil {
-		logger.Info("Error syncing the Channel config", zap.Error(err))
-		r.recorder.Eventf(c, corev1.EventTypeWarning, channelConfigSyncFailed, "Failed to sync Channel config: %v", err)
-		return err
-	}
-
 	if c.DeletionTimestamp != nil {
-		// K8s garbage collection will delete the K8s service and VirtualService for this channel.
+		// K8s garbage collection will delete the K8s service for this channel.
 		// We use a finalizer to ensure the channel config has been synced.
 		util.RemoveFinalizer(c, finalizerName)
 		return nil
 	}
 
 	util.AddFinalizer(c, finalizerName)
 
-	svc, err := util.CreateK8sService(ctx, r.client, c)
+	svc, err := util.CreateK8sService(ctx, r.client, c, util.ExternalService(c))
```

Comment:
There is an issue here. Existing K8s service of type: ClusterIP  cannot be updated to type:ExternalName. Hence this will not work for eventing upgrades from 0.5 to 0.6. Will fix this in next iteration and update this PR.